RELEASES.md

```diff
@@ -0,0 +1,87 @@
+# Releases
```

Comment:
Maybe we could add back in `1. [Compatibility guarantees](#compatibility-guarantees)` but just with a link to `VERSIONING.md`?

Reply:
Done in https://github.com/timberio/vector/pull/8795/commits/ac1a0559f4b725513f921a5e4883c31fb1e25d97